jax/_src/numpy/linalg.py

```diff
@@ -2136,3 +2136,52 @@ def cond(x: ArrayLike, p=None):
     r = norm(x, ord=p, axis=(-2, -1)) * norm(inv(x), ord=p, axis=(-2, -1))
   # Convert NaNs to infs where original array has no NaNs.
   return jnp.where(ufuncs.isnan(r) & ~ufuncs.isnan(x).any(axis=(-2, -1)), jnp.inf, r)
+
+
+def trace(x: ArrayLike, /, *,
+          offset: int = 0, dtype: DTypeLike | None = None) -> Array:
+  """Compute the trace of a matrix.
+
+  JAX implementation of :func:`numpy.linalg.trace`.
+
+  Args:
+    x: array of shape ``(..., M, N)`` and whose innermost two
+      dimensions form MxN matrices for which to take the trace.
+    offset: positive or negative offset from the main diagonal
+      (default: 0).
+    dtype: data type of the returned array (default: ``None``). If ``None``,
+      the returned array will have the same data type as ``x``. If ``x`` is of
+      integer type of precision less than the default integer precision,
+      then the default integer precision is used.
+
+  Returns:
+    array containing the traces and whose shape is determined by
```

Comment:
I think it would be simpler to say something like `array of batched traces, shape = x.shape[:-2]`